compiler/rustc_span/src/lib.rs

```diff
@@ -1194,6 +1194,28 @@ impl SourceFileHash {
     }
 }
 
+#[derive(HashStable_Generic)]
+#[derive(Copy, PartialEq, PartialOrd, Clone, Ord, Eq, Hash, Debug, Encodable, Decodable)]
+pub enum DebuggerVisualizerType {
+    Natvis,
+}
+
+/// A single debugger visualizer file.
+#[derive(HashStable_Generic)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Encodable, Decodable)]
+pub struct DebuggerVisualizerFile {
+    /// The complete debugger visualizer source.
+    pub src: Vec<u8>,
```

Comment:
With the current version, cloning a `DebuggerVisualizerFile` will clone the entire `Vec<u8>`. I suggest making `src` an `Arc<[u8]>`, so that cloning will just clone the reference-counted pointer instead of doing a deep copy. See https://doc.rust-lang.org/std/vec/struct.Vec.html#example-1 for an example.